codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -334,22 +346,36 @@ class JsonSerializerGenerator(
 
     private fun RustWriter.serializeMember(context: MemberContext) {
         val targetShape = model.expectShape(context.shape.target)
-        if (symbolProvider.toSymbol(context.shape).isOptional()) {
-            safeName().also { local ->
+        safeName().also { local ->
```

Comment:
nit: the unconditional call to `safeName()` here is increasing the counter regardless of whether `local` is used or not, hence why sometimes the codegen diff is showing that we "skip" a variable name.

Reply:
[7d4d59f](https://github.com/awslabs/smithy-rs/pull/2005/commits/7d4d59fad23f712f7c37ef6582b09e7330717eaf) should fix it 👍🏻